include/materials/ConcreteASREigenstrain.h

```diff
@@ -0,0 +1,109 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/*                       BlackBear                              */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef CONCRETEASREIGENSTRAIN_H
+#define CONCRETEASREIGENSTRAIN_H
+
+#include "ConcreteExpansionEigenstrainBase.h"
+#include "RankTwoTensor.h"
+
+/**
+ * Computes the volumetric expansion eigenstrain due to alkali-silica reaction.
+ */
+
+class ConcreteASREigenstrain : public ConcreteExpansionEigenstrainBase
+{
+public:
+  ConcreteASREigenstrain(const InputParameters & parameters);
+
+  void initQpStatefulProperties() override;
+  bool needStressEigenvalues() override { return true; }
+  Real computeQpVolumetricStrain() override;
+
+  /**
+   * Compute the residual for the computation of the ASR reaction extent.
+   * @param qp Current quadrature point id
+   * @param scalar Current iterative value of the ASR reaction extent
+   * @return residual, which is the error in the new ASR reaction extent
+   */
+  Real computeResidual(unsigned qp, Real scalar);
+
+  /**
+   * Compute the derivative of the residual for the ASR reaction extent.
+   * @param qp Current quadrature point id
+   * @param scalar Current iterative value of the ASR reaction extent
+   * @return derivative, which is the change in the error in the new ASR
+   *         reaction extent wrt the reaction extent.
+   */
+  Real computeDerivative(unsigned qp, Real scalar);
+
+private:
+  /// Coupled value of temperature
+  const VariableValue & _temperature;
+
+  /// Coupled value of relative humidity
+  const VariableValue & _rh;
+  /// Power to which relative humidity is raised
+  const Real & _rh_exponent;
```

Comment:
What's going on with all these abbreviations? Wouldn't `_relative_humidity` be a much nicer variable name?